packages/opentelemetry-plugin-document-load/test/documentLoad.test.ts

```diff
@@ -0,0 +1,282 @@
+/*!
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as assert from 'assert';
+import * as sinon from 'sinon';
+
+import { ConsoleLogger } from '@opentelemetry/core';
+import {
+  BasicTracer,
+  ReadableSpan,
+  SimpleSpanProcessor,
+  SpanExporter,
+} from '@opentelemetry/tracing';
+import { Logger, PluginConfig } from '@opentelemetry/types';
+
+import { ExportResult } from '../../opentelemetry-base/build/src';
+import { DocumentLoad } from '../src';
+
+export class DummyExporter implements SpanExporter {
+  export(
+    spans: ReadableSpan[],
+    resultCallback: (result: ExportResult) => void
+  ) {}
+
+  shutdown() {}
+}
+
+describe('DocumentLoad Plugin', () => {
+  let plugin: DocumentLoad;
+  let moduleExports: any;
+  let tracer: BasicTracer;
+  let logger: Logger;
+  let config: PluginConfig;
+  let spanProcessor: SimpleSpanProcessor;
+  let dummyExporter: DummyExporter;
+
+  beforeEach(() => {
+    Object.defineProperty(window.document, 'readyState', {
+      writable: true,
+      value: 'complete',
+    });
+    moduleExports = {};
+    tracer = new BasicTracer();
+    logger = new ConsoleLogger();
+    config = {};
+    plugin = new DocumentLoad();
+    dummyExporter = new DummyExporter();
+    spanProcessor = new SimpleSpanProcessor(dummyExporter);
+    tracer.addSpanProcessor(spanProcessor);
+  });
+
+  afterEach(() => {
+    Object.defineProperty(window.document, 'readyState', {
+      writable: true,
+      value: 'complete',
+    });
+  });
+
+  describe('constructor', () => {
+    it('should construct an instance', () => {
+      plugin = new DocumentLoad();
+      assert.ok(plugin instanceof DocumentLoad);
+    });
+  });
+
+  describe('when document readyState is complete', () => {
+    it('should start collecting the performance immediately', () => {
+      const spy_OnDocumentLoaded = sinon.spy(
+        plugin,
+        '_collectPerformance' as never
+      );
+      plugin.enable(moduleExports, tracer, logger, config);
+      assert.strictEqual(window.document.readyState, 'complete');
+      assert.ok(spy_OnDocumentLoaded.calledOnce);
```

Comment:
Rather than testing that the private function gets called, would it make sense instead to check that spans are actually written? I generally encourage only testing the public interface of a class.